src/librustc/ty/structural_impls.rs

```diff
@@ -58,6 +59,13 @@ impl<'tcx, T: Lift<'tcx>, E: Lift<'tcx>> Lift<'tcx> for Result<T, E> {
     }
 }
 
+impl<'tcx, T: Lift<'tcx>> Lift<'tcx> for Box<T> {
```

Comment:
This feels suspicious. Why do we have boxes of pointers? Maybe I'm not understanding something...